stream_provider.go

```diff
@@ -254,48 +259,77 @@ func ParseStreamName(fullStreamName string) (string, string) {
 	return fullStreamName, ""
 }
 
-func (r *streamRegistry) register(streamName, dataType string, p *StreamProvider) {
+type StreamDefinitions []*StreamDefinition
+
+func (d StreamDefinitions) StreamNames() string {
+	return d.join(func(sd *StreamDefinition) string {
+		return sd.streamName
+	})
+}
+
+func (d StreamDefinitions) join(f func(*StreamDefinition) string) string {
+	l := make([]string, len(d))
+	for i, v := range d {
+		l[i] = f(v)
+	}
+	return strings.Join(l, ",")
+}
+
+func (d StreamDefinitions) DataTypes() string {
+	return d.join(func(sd *StreamDefinition) string {
+		return sd.dataType
+	})
+}
+
+func (r *streamRegistry) streams() StreamDefinitions {
+	r.Lock()
+	defer r.Unlock()
+	return extractStreamsDefinitions(r)
+}
+
+//To be called only with a lock on the stream registry
+func extractStreamsDefinitions(r *streamRegistry) StreamDefinitions {
+	result := make([]*StreamDefinition, len(r.providers))
+	i := 0
+	for _, v := range r.providers {
```

Comment:
you can directly use:

`for i,v := range r.providers{
 }`
so you don't need to increment i by hand

Reply:
r.providers is a map, to in this case i would be the key, wouldn't it?